File: src/RuleEngine/Operator.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace RuleEngine
{
    public class Operator : IRuleElement
    {
        public Operator(Operators @operator)
        {
            Name = @operator.ToString();
            Value = @operator;                
        }

        public Guid Id { get; set; } = Guid.NewGuid();
        public string Name { get; set; }
        public object Value { get; set; }

        public bool Process(IRuleElement element, Operator @operator)
        {
            throw new InvalidOperationException("This operation is not available on this rule element.");
        }
    }
}

File: src/RulesEngine.WebApp/Scripts/rules-engine/Operator.ts
import { IRuleElement } from "./IRuleElement"
import { Operators } from "./Operators"

export class Operator implements IRuleElement {
    public constructor(operator: Operators) {
        this.Name = operator.toString();
        this.Value = operator;
    }

    public Id: string = "";
    public Name: string = "";
    public Value: any = {};

    Process(element: IRuleElement, operator: Operator): boolean {
        throw new TypeError("This operation is not available on this rule element.");
    }
}